newsAPI.js

```diff
@@ -1,17 +1,92 @@
-var generateArticle = function(response) {
-
-    var url = 'https://newsapi.org/v2/everything?' +
-          'q=Apple&' +
-          'from=2018-03-28&' +
-          'sortBy=popularity&' +
-          'apiKey=0873dd38116a4b1d9db9c7f2d99754a7';
-
-    var req = new Request(url);
-
-    var article = fetch(req)
-        .then(function(response) {
-            console.log(response.json());
-        return article;
-        })
-};
-module.exports = generateArticle;
+
+$(document).ready(()=>{
+	getNews();
+
+	function getNews()
+	{
+		let endPoint = "https://newsapi.org/v1/articles";
+		let apiKey = "0873dd38116a4b1d9db9c7f2d99754a7";
+		//sources can be added from the included links.txt
+		let urls = [
+			`${endPoint}?source=engadget&sortBy=latest&apiKey=${apiKey} `,
+			`${endPoint}?source=fortune&sortBy=latest&apiKey=${apiKey} `
+		];
+
+		let allResults = [];
+
+		let count = urls.length-1;
+		const get =(real)=>{
+			$.getJSON(urls[ count ], function(data) {
+				console.log("JSON data has been retrieved from " + data.source);
+				let news = data.articles; //get only the news articles
+				allResults.push(news)
+				// printNews(news);
+				real();
+			})
+		};
+		recurse();		
+
+		function recurse(){
+			if(count >= 0){
+				get(recurse);
+				count--;
+			}
+			else
+				//allResults is  an arrray of nested objects
+				printNews(allResults);
+		}
+	}
+
+	//display the news
+	function printNews(result)
+	{	
+
+		let res=[];
+		//flatten the array of nested objects into one single array
+		result.map(list=>{
+			// console.log(list)
+			return list.map(item=>{
+				// console.log(item)
+				res.push(item)
+			})
+		})
+		console.log(res)
+		//Shuffle all the news items
+		shuffleArray(res);		
+		let output = "";
+		for(let i = 0; i < res.length; i++)
+		{
+
+			let link =  res[i].url;
+			let resultDiv = `
+				<div class="col-sm-4 col-md-4">
+					<div class="thumbnail">
+						<img src="${res[i].urlToImage}" alt="${res[i].title}" class="img-responsive">
+						<div class="caption">
+							<h2> ${res[i].title} </h2>
+							<h4> ${res[i].description} </h4>
+							<p><a href="${link}" target="_blank" class="btn btn-primary" role="button">View Article</a> </p>
+							<p>
+								<a href="${link}" target="_blank" class="btn btn-primary" role="button">Fair</a>
+								<a href="${link}" target="_blank" class="btn btn-primary" role="button">Unfair</a>  
+								<a href="${link}" target="_blank" class="btn btn-primary" role="button">Newsworthy</a> 
+								<a href="${link}" target="_blank" class="btn btn-primary" role="button">Not Newsworthy</a> 
+							</p>
+						</div>
+					</div>
+				</div>	`
+			output += resultDiv;
+		}
+		$('.printResults').html(output);
```

Comment:
This line scares me.  Rather than setting the inner html to dom-like elements.  Would it be possible to refactor to append dom elements and then set the text content?

Reply:
This is a good suggestion. I would like to try and refactor as time permits.